src/getViewForSwagger2.ts

```diff
@@ -116,182 +134,154 @@ export function getViewForSwagger2(opts: CodeGenOptions): ViewData{
+    fp.filter(isParameters),
+    fp.entries
+)(api);
 
-            if(method.isSecure && method.isSecureApiKey) {
-                data.isSecureApiKey = method.isSecureApiKey;
-            }
+const authorizedMethods = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'COPY', 'HEAD', 'OPTIONS', 'LINK', 'UNLINK', 'PURGE', 'LOCK', 'UNLOCK', 'PROPFIND'];
+const isAuthorizedMethod = (method: string) => authorizedMethods.indexOf(method.toUpperCase()) > -1 
 
-            if(method.isSecure && method.isSecureBasic) {
-                data.isSecureBasic = method.isSecureBasic;
-            }
+function getIntVersion(path: string): number {
+    return parseInt(getVersion(path).substr(1));
+}
 
-            const produces = op.produces || swagger.produces;
-            if(produces) {
-                method.headers.push({
-                  name: 'Accept',
-                  value: `'${produces.join(', ')}'`,
-                });
+// TODO: Do some typescript magic with values from authorizedMethods
+type EndpointDescriptor = [string, HttpEndpointDescriptor];
+type HttpEndpointDescriptor = { [index: string]: HttpOperation};
+type HttpOperationEntryWithGlobalParamsAndPath = [string, string, HttpOperation, ReadonlyArray<Parameter>];
+
+const addPathAndGlobalParams = ([path, api]: EndpointDescriptor): HttpOperationEntryWithGlobalParamsAndPath[] => fp.entries(api).map(([httpVerb, httpOperationDescription]): HttpOperationEntryWithGlobalParamsAndPath => [path, httpVerb, httpOperationDescription, getGlobalParams(api)])
+const addGlobalParamsToEntries = (endpointDescriptor: EndpointDescriptor) => addPathAndGlobalParams(endpointDescriptor);
+
+// TODO: Remind me to clean this method :P
+const makeMethodsFromPaths = (data: ViewData, opts: CodeGenOptions, swagger: Swagger): Method[] => fp.flatten(
+    fp.flatten(fp.entries(swagger.paths).map(addGlobalParamsToEntries))
+        .filter(([_path, httpVerb]) => isAuthorizedMethod(httpVerb))
+        // Ignore deprecated endpoints
+        .filter(([_path, _httpVerb, op]) => !op.deprecated)
+        .map(([path, httpVerb, op, globalParams]) => {
+        // TODO: Start of untested security stuff that needs fixing
+        const secureTypes = [];
+
+        if(swagger.securityDefinitions !== undefined || op.security !== undefined) {
+            const mergedSecurity = merge([], swagger.security, op.security).map((security) => {
+                return Object.keys(security);
+            });
+            if(swagger.securityDefinitions) {
+                for(const sk in swagger.securityDefinitions) {
+                    if (mergedSecurity.join(',').indexOf(sk) !== -1){
+                        secureTypes.push(swagger.securityDefinitions[sk].type);
+                    }
+                }
             }
+        }
+        // End of untested 
+
+        const method: Method = makeMethod(path, opts, swagger, httpVerb, op, secureTypes, globalParams);
+
+        // TODO: It seems the if statements below are pretty weird... 
```

Comment:
Definitely something that should be looked into and cleaned up at some point. Dont have to do that now ofc

Reply:
Should be part of the same security issue i guess #54